### DISCUSSION TITLE
How to trigger a GitHub Action workflow with Directus Webhooks

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#12591] [Archive]
We want to trigger a GitHub Action workflow (to build an static site) with Directus Webhooks.

There are any guide on this?

Thanks!

**Comment by abdonrd** [comment]
I just find this interesting link:

https://docs.github.com/en/rest/reference/repos#create-a-repository-dispatch-event

**Reply by rijkvanzanten** [reply]
I think you just answered your own question!

**Reply by abdonrd** [reply]
How we can send the `event_type` in the body? It's a required parameter.

**Reply by rijkvanzanten** [reply]
You can't configure that in the current implementation of webhooks. That being said, https://github.com/directus/directus/pull/12522 has support for custom data structures in headers/body 🙂

**Reply by abdonrd** [reply]
Ouch! Then we can't connect GitHub Actions with Directus Webhooks. 😕

The Flows will replace the Webhooks?

**Reply by fitu86** [reply]
Check this https://discord.com/channels/725371605378924594/741319150030553159/961052915806642196

**Reply by abdonrd** [reply]
> Check this https://discord.com/channels/725371605378924594/741319150030553159/961052915806642196

I didn't fully understand the video.
I haven't seen any reference to [Directus Webhooks](https://docs.directus.io/configuration/webhooks/) or GitHub Actions.

**Reply by rijkvanzanten** [reply]
> Ouch! Then we can't connect GitHub Actions with Directus Webhooks. 😕
> 
> The Flows will replace the Webhooks?

Flows will replace webhooks, yeah! There's a "trigger http call" operation in Flows that offers feature parity with the existing webhooks setup, and that will also allow you to configure headers/body of the request 👍🏻

**Comment by Horsty80** [comment]
@abdonrd hi ! I don't know if you still need help but I answer to add sample for other who asking for the same questions. (I one of them :p)

Flow is actually the best way to achieve this.

<img width="686" alt="Capture d’écran 2023-01-05 à 01 17 11" src="https://user-images.githubusercontent.com/15615569/210673953-8b3ba48d-4900-4f2d-9d0e-b1734a4f3ba2.png">

## Create the trigger card
1) create a trigger , in my case "Manual".
2) Setup where to find the button (in case of manual trigger I choose to add it in my collection `Post`)
2) Save

## Create the web hook card

<img width="1140" alt="Capture d’écran 2023-01-05 à 01 23 45" src="https://user-images.githubusercontent.com/15615569/210674636-9ebdbbe4-49bd-4a18-9b3c-8468ace393d8.png">

1) Click on the <kbd>+</kbd> on right border of the trigger card
2) Select `webhook / request url`
3) Fill the form with our data (ex: url to trigger repository dispatch: `https://api.github.com/repos/owner/repos_name/dispatches`)
4) Add required header in the header section

⚡️ Don't forget to create a PAT in your GitHub setting -> developers settings ⚡️

```
Accept: application/vnd.github.v3+json
Content-Type: application/json
Authorization: token your_generated_PAT_with_good_right
```
5) Add Json body with `event_type` property to trigger expected workflow
6) Save

Now you can go to the configured collection and expand right menu `flow` and you can find our new Button to manual trigger the Flow

It's simply basic approach how to use flow and GitHub action. It's a good start for customization for your need.
<img width="503" alt="Capture d’écran 2023-01-05 à 01 29 05" src="https://user-images.githubusercontent.com/15615569/210675127-d8447ffd-260f-4d9e-bacc-ca5b6e2b1b4c.png">
